src/main/java/HttpServer/RequestParser.java

```diff
@@ -2,26 +2,47 @@
 
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.HashMap;
 
 public class RequestParser {
 
-    private ArrayList<String> requestLines; // Associative data structure?
-    private Readable reading;
+    private ArrayList<String> lines;
+    public HashMap request;
+    private Readable source;
 
-    public RequestParser(Readable reading) {
-        this.requestLines = new ArrayList<>();
-        this.reading = reading;
+    public RequestParser(Readable source) {
+        this.lines = new ArrayList<>();
+        this.request = new HashMap();
+        this.source = source;
     }
 
     public void read() throws IOException {
         System.out.println("Reading request: ");
         System.out.println("--------------------------");
-        String line = reading.readLine();
+        String line = source.readLine();
```

Comment:
Keep in mind that you want to make sure you have no extraneous output when running tests. Any such outputs end up being noise and making it harder to see real errors/warning/failures/etc.

If you want to have a logger, which is a good thing, there are strategies to do so without polluting STDOUT when running tests.

Reply:
:white_check_mark: